### PR TITLE
Fix ORT build when onnxruntime_PYBIND_EXPORT_OPSCHEMA is enabled

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: ORT doesn't build when onnxruntime_PYBIND_EXPORT_OPSCHEMA  is enabled (when doc generation is enabled). This change fixes it. It also enables doc generation in one of the required CI checks so that any regression in doc generation is caught.

### CODE REVIEW

File: onnxruntime/python/onnxruntime_pybind_state.cc [hunk 1]
```diff
@@ -307,9 +307,6 @@ void RegisterExecutionProviders(InferenceSession* sess, const std::vector<std::s
     } else if (type == kCudaExecutionProvider) {
 #ifdef USE_CUDA
       RegisterExecutionProvider(sess, *onnxruntime::CreateExecutionProviderFactory_CUDA(cuda_device_id, cuda_mem_limit, arena_extend_strategy));
-      cuda_device_id = 0;
```
Comment (hariharans29) [inline onnxruntime/python/onnxruntime_pybind_state.cc:310]: This is already set at a global level and used in the previous line. There is no need to reset these values again.

File: onnxruntime/python/onnxruntime_pybind_state.cc [hunk 2]
```diff
@@ -472,7 +469,8 @@ void addOpSchemaSubmodule(py::module& m) {
   auto schemadef = m.def_submodule("schemadef");
   schemadef.doc() = "Schema submodule";
 
-  py::class_<ONNX_NAMESPACE::OpSchema> op_schema(schemadef, "OpSchema");
+  // Keep this binding local to this module
+  py::class_<ONNX_NAMESPACE::OpSchema> op_schema(schemadef, "OpSchema", py::module_local());
```
Comment (hariharans29) [inline onnxruntime/python/onnxruntime_pybind_state.cc:473]: Without these, the CI build fails as it seems these binding definitions clash with the ONNX Python bindings for the ONNX classes. hence, kept these bindings local to this module. (Not sure why this doesn't repro locally, but these bindings can be local to the onnxruntime module and that solves it)

File: onnxruntime/python/onnxruntime_pybind_state.cc [hunk 3]
```diff
@@ -403,7 +400,7 @@ void addGlobalMethods(py::module& m, const Environment& env) {
         std::vector<std::shared_ptr<onnxruntime::IExecutionProviderFactory>> factories = {
             onnxruntime::CreateExecutionProviderFactory_CPU(0),
 #ifdef USE_CUDA
-            onnxruntime::CreateExecutionProviderFactory_CUDA(0),
+            onnxruntime::CreateExecutionProviderFactory_CUDA(cuda_device_id, cuda_mem_limit, arena_extend_strategy),
 #endif
```
Comment (skottmckay) [inline onnxruntime/python/onnxruntime_pybind_state.cc:404]: This change probably isn't necessary as all the EPs here are created with default args as the instances are only used to get the kernel defs.

Comment (hariharans29) [reply onnxruntime/python/onnxruntime_pybind_state.cc:404]: This was the core change in this PR. This was causing the build break. I think previously some arguments had default values. Now, we need to explicitly provide them.

Comment (snnn) [reply onnxruntime/python/onnxruntime_pybind_state.cc:440]: Oh, the training PR changed the signature. But we'll never be able to do the corresponding change on the C API, even though the  CreateExecutionProviderFactory_CUDA function was created for the C APIs.
We should support multiple GPU devices. Hector Li has a test pipeline for  that feature and it runs daily. But as you can see, as long as we hard code the '0' here. we can't really support multiple GPUs through the python interface.

Comment (hariharans29) [reply onnxruntime/python/onnxruntime_pybind_state.cc:440]: yes, the training PR changed the signature and this change basically accommodates that.  I would think that for the the doc gen exercise, the intricacies involved with different GPU devices should not matter given that we only create the CUDA EP to query its kernel registry for the sake of creating the documentation (please correct me if I am wrong). And you are right, here the change is equivalent to hard-coding '0' for the device id.

**Pull request merged**